pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -725,6 +726,22 @@ func (adc *attachDetachController) processVolumeAttachments() error {
 				err)
 			continue
 		}
+		inTreePluginName := plugin.GetPluginName()
+		if adc.csiMigratedPluginManager.IsMigrationEnabledForPlugin(inTreePluginName) {
+			plugin, _ = adc.volumePluginMgr.FindAttachablePluginByName(csi.CSIPluginName)
+			// podNamespace is not needed here for Azurefile as the volumeName generated will be the same
+			volumeSpec, err = csimigration.TranslateInTreeSpecToCSI(volumeSpec, "" /* podNamespace */, adc.intreeToCSITranslator)
+			if err != nil {
+				klog.Errorf(
+					"Failed to translate intree volumeSpec to CSI volumeSpec for volume:%q, va.Name:%q, nodeName:%q: %v",
+					*pvName,
+					va.Name,
+					nodeName,
+					inTreePluginName,
+					err)
+				continue
```

Comment:
I think it should be possible to cover this via an unit test in `Test_ADC_VolumeAttachmentRecovery`.

Reply:
@codablock Can you help to add a unit test case here to verify it is working as expected? If it end up being too complicated, we can follow up in the next PR.